backend/src/apiserver/server/job_server.go

```diff
@@ -81,394 +80,305 @@ type JobServer struct {
-}
-
-func (s *JobServer) DisableJob(ctx context.Context, request *apiv1beta1.DisableJobRequest) (*empty.Empty, error) {
-	if s.options.CollectMetrics {
-		disableJobRequests.Inc()
+		return nil, util.Wrap(err, "Failed to list v1beta1 runs: validating filter failed")
+	}
+	namespace := ""
+	experimentId := ""
+	if filterContext.ReferenceKey != nil {
+		switch filterContext.ReferenceKey.Type {
+		case model.NamespaceResourceType:
+			namespace = filterContext.ReferenceKey.ID
+		case model.ExperimentResourceType:
+			experimentId = filterContext.ReferenceKey.ID
+		}
 	}
-
-	err := s.canAccessJob(ctx, request.Id, &authorizationv1.ResourceAttributes{Verb: common.RbacResourceVerbDisable})
+	jobs, total_size, nextPageToken, err := s.listJobs(ctx, r.GetPageToken(), int(r.GetPageSize()), r.GetSortBy(), r.GetFilter(), namespace, experimentId)
 	if err != nil {
-		return nil, util.Wrap(err, "Failed to authorize the request")
+		return nil, util.Wrap(err, "Failed to list jobs")
 	}
-
-	return s.enableJob(ctx, request.Id, false)
+	apiJobs := toApiJobsV1(jobs)
+	if apiJobs == nil {
+		return nil, util.NewInternalServerError(util.NewInvalidInputError("Failed to convert internal recurring run representations to their v1beta1 API counterparts"), "Failed to list v1beta1 recurring runs")
+	}
+	return &apiv1beta1.ListJobsResponse{
+		Jobs:          apiJobs,
+		TotalSize:     int32(total_size),
+		NextPageToken: nextPageToken,
+	}, nil
 }
 
-func (s *JobServer) DeleteJob(ctx context.Context, request *apiv1beta1.DeleteJobRequest) (*empty.Empty, error) {
+func (s *JobServer) EnableJob(ctx context.Context, request *apiv1beta1.EnableJobRequest) (*empty.Empty, error) {
 	if s.options.CollectMetrics {
-		deleteJobRequests.Inc()
+		enableJobRequests.Inc()
 	}
-
-	err := s.canAccessJob(ctx, request.Id, &authorizationv1.ResourceAttributes{Verb: common.RbacResourceVerbDelete})
+	err := s.enableJob(ctx, request.GetId())
 	if err != nil {
-		return nil, util.Wrap(err, "Failed to authorize the request")
+		return nil, util.Wrap(err, "Failed to enable a v1beta1 recurring run")
 	}
+	return &empty.Empty{}, nil
+}
 
-	err = s.resourceManager.DeleteJob(ctx, request.Id)
+func (s *JobServer) disableJob(ctx context.Context, jobId string) error {
+	err := s.canAccessJob(ctx, jobId, &authorizationv1.ResourceAttributes{Verb: common.RbacResourceVerbEnable})
 	if err != nil {
-		return nil, err
+		return util.Wrap(err, "Failed to authorize the request")
 	}
+	return s.resourceManager.ChangeJobMode(ctx, jobId, false)
+}
 
+func (s *JobServer) DisableJob(ctx context.Context, request *apiv1beta1.DisableJobRequest) (*empty.Empty, error) {
 	if s.options.CollectMetrics {
-		jobCount.Dec()
+		disableJobRequests.Inc()
+	}
+
+	err := s.disableJob(ctx, request.GetId())
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to disable a v1beta1 recurring run")
 	}
 	return &empty.Empty{}, nil
 }
 
-func (s *JobServer) validateCreateJobRequest(request *apiv1beta1.CreateJobRequest) error {
```

Comment:
why removing the validate requests method? or was this moved to a different location?

Reply:
There is no longer need to validate the request. A part of it sits in `toModelJob` and the rest is in the `createJob`. This is to apply the same validation for v1 and v2 messages.